src/lib.rs

```diff
@@ -1444,6 +1476,43 @@ impl<'a, U: 'a + ?Sized + Borrow<str>, I: Clone + IntoIterator<Item=&'a U>> Sour
         }
         DoesNotMatch
     }
+    /// https://www.w3.org/TR/CSP/#match-integrity-metadata-to-source-list
+    fn does_integrity_metadata_match_source_list(&self, integrity_metadata: &str) -> MatchResult {
```

Comment:
Based on the results of `tests/wpt/tests/content-security-policy/default-src/default-src-sri_hash.sub.html` I still think something is off with this implementation. But at least it now passes `tests/wpt/tests/content-security-policy/script-src/script-src-report-only-policy-works-with-external-hash-policy.html` which I consider more important